Source/Scene/LabelCollection.js

```diff
@@ -267,100 +270,136 @@ define([
         label._repositionAllGlyphs = true;
     }
 
+    function calculateWidthOffset(lineWidth, horizontalOrigin, backgroundPadding)
+    {
+        if (horizontalOrigin === HorizontalOrigin.CENTER) {
+            return -lineWidth / 2;
+        } else if (horizontalOrigin === HorizontalOrigin.RIGHT) {
+            return -(lineWidth + backgroundPadding.x);
+        }
+        return backgroundPadding.x;
+    }
+
     // reusable Cartesian2 instances
     var glyphPixelOffset = new Cartesian2();
     var scratchBackgroundPadding = new Cartesian2();
 
     function repositionAllGlyphs(label, resolutionScale) {
         var glyphs = label._glyphs;
+        var text = label._text;
         var glyph;
         var dimensions;
-        var totalWidth = 0;
-        var maxDescent = Number.NEGATIVE_INFINITY;
-        var maxY = 0;
+        var lastLineWidth = 0;
+        var maxLineWidth = 0;
+        var lineWidths = [];
```

Comment:
My hope is we don't need a scratch in this case.  This function only gets called when the glyphs all need to be repositioned, which is an expensive operation we try to avoid (using the delay tactics you pointed out in the label background PR).